pkg/cluster/install.go

```diff
@@ -57,109 +94,73 @@ func (m *manager) adminUpdate() []steps.Step {
 		// struct, so we'll rebuild the fpAuthorizer and use the error catching
 		// to advance
 		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.fixupClusterSPObjectID),
-		steps.Action(m.fixInfraID), // Old clusters lacks infraID in the database. Which makes code prone to errors.
+		steps.Action(m.startVMs),
+		steps.Condition(m.apiServersReady, 30*time.Minute, true),
 	}
 
-	if isEverything {
-		toRun = append(toRun,
-			steps.Action(m.ensureResourceGroup), // re-create RP RBAC if needed after tenant migration
-			steps.Action(m.createOrUpdateDenyAssignment),
-			steps.Action(m.ensureServiceEndpoints),
-			steps.Action(m.populateRegistryStorageAccountName), // must go before migrateStorageAccounts
-			steps.Action(m.migrateStorageAccounts),
-			steps.Action(m.fixSSH),
-			// steps.Action(m.removePrivateDNSZone), // TODO(mj): re-enable once we communicate this out
-		)
+	// Generic fix-up actions that are fairly safe to always take, and don't require a running cluster
```

Comment:
Curious about why the `fixInfraID` action is outside the list above. The actions in that list also comply with the description of the comment for this action (in the old code, the comment was actually for all of them).

Reply:
My thinking was that before there were several generic tasks that happened for every path. These were all clearly bootstrapping tasks apart from this one which is actually changing the state of the cluster.